Improved memory handling (#14)
eriksen-lab/opentrustregion: README.md

- **`local_red_factor`** (real): Reduction factor for the residual during micro iterations in the local region.
- **`verbose`** (integer): Controls the verbosity of output during optimization.
- **`seed`** (integer): Seed value for generating random trial vectors.
- **`logger`** (subroutine): Accepts a log message. Logging is otherwise routed to stdout.

## Stability Check

> A separate `stability_check` subroutine is available to verify whether the current solution corresponds to a minimum. If not, it returns a boolean indicating instability and optionally an additional direction along the eigenvector corresponding to the negative eigenvalue.

Reviewer: isn't this a function?

Reviewer: and optionally, writes the eigenvector in-place to the provided memory

Author: No, it is a subroutine that returns the stability, error code, and optional eigenvector as intent(out) arguments. I have changed the README in correspondence with your second comment.

### Required Arguments

- **`h_diag`** (real array): Represents the Hessian diagonal at the current point.
- **`hess_x`** (subroutine): Performs Hessian-vector products at the current point:
  - Accepts a trial vector and writes the result of the Hessian transformation into an output array
  - Returns an integer error code (0 for success, positive integers < 100 for recoverable errors)

> - **`stable`** (boolean): Returns whether the current point is stable.

Reviewer: nvm it's listed below

- **`error`** (integer): An integer code indicating the success or failure of the solver. The error code structure is explained below.

### Optional Arguments

> - **`kappa`** (real array): If the memory is provided, returns descent direction if current point is not stable otherwise returns zeros.

Reviewer: ```suggestion
- **`kappa`** (real array): If the memory is provided and the current point is not stable (as checked return code of `stable`), returns descent direction.
```
